src/eth/types.ts

```diff
@@ -11,3 +11,8 @@ export const EthUint256Struct = definePattern(
   'EthUint256',
   /^0x([1-9a-f][0-9a-f]*|0)$/iu,
 );
+
+export const BundlerUrlStruct = definePattern(
+  'BundlerUrl',
+  /^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\\+.~#?&\\/=]*)$/iu,
```

Comment:
Do you have a source for this regex? I suspect that it might not be completely right, and I've seen quite a few variations.

I wonder if it wouldn't be simpler to add a validation function like [this one](https://stackoverflow.com/a/43467144):

```javascript
function isValidHttpUrl(string) {
  let url;
  
  try {
    url = new URL(string);
  } catch (_) {
    return false;  
  }

  return url.protocol === "http:" || url.protocol === "https:";
}
```

Reply:
Yea that makes sense, and is simpler